src/Client.php

```diff
@@ -1507,6 +1509,17 @@ private function getErrorCodesFromXmlElement(SimpleXMLElement $errors): array {
         return $errorCodes;
     }
 
+    /**
+     * Sets the provided date to either null or a DateTimeInterface.
+     */
+    private function toDateTimeInterface(string $dateString): DateTimeInterface|null {
+        if ($dateString === null || $dateString === '') {
```

Comment:
`$dateString` can never be null, PHP will raise a `TypeError` because this file begins with  `declare(strict_types=1);` and you have declared the parameter to be a string on the line above. That's fine because there is always an `if (isset($report->PropertyName)) {` around where you use this method. So I think we ought to simplify just a bit more to:
```suggestion
        if ($dateString === '') {
```